web-app/packages/lib/src/modules/project/views/ProjectVersionsViewTemplate.vue

```diff
@@ -0,0 +1,293 @@
+        v-model="options.page"
+        :length="Math.ceil(versionsCount / options.itemsPerPage)"
+        :total-visible="7"
+        circle
+        color="primary"
+        @input="fetchPage"
+      ></v-pagination>
+    </div>
+    <slot name="table-footer"></slot>
+  </div>
+</template>
+
+<script lang="ts">
+import Vue, { PropType } from 'vue'
+import { mapActions, mapState } from 'vuex'
+
+import MerginAPIMixin from '@/common/mixins/MerginAPIMixin'
+import {
+  FetchProjectVersionsParams,
+  ProjectVersion,
+  ProjectVersionsItem
+} from '@/modules'
+
+export default Vue.extend({
+  name: 'ProjectVersionsViewTemplate',
+  mixins: [MerginAPIMixin],
+  props: {
+    projectName: String,
+    namespace: String,
+    asAdmin: {
+      type: Boolean,
+      default: false
+    },
+    /** Default items per page */
+    defaultItemsPerPage: Number as PropType<number>,
+    /** Disabled keys (name attribute of rows in vuetify table are keys for items) */
+    disabledKeys: { type: Array as PropType<string[]>, default: () => [] },
+    /** Show pagination */
+    showPagination: { type: Boolean, default: true }
+  },
+  data() {
+    return {
+      headers: [
+        { text: 'Version', value: 'name' },
+        { text: 'Created', value: 'created' },
+        { text: 'Author', value: 'author', sortable: false },
+        {
+          text: 'Added',
+          value: 'changes.added',
+          icon: 'add_circle',
+          color: 'green',
+          sortable: false
+        },
+        {
+          text: 'Removed',
+          value: 'changes.removed',
+          icon: 'delete',
+          color: 'red',
+          sortable: false
+        },
+        {
+          text: 'Modified',
+          value: 'changes.updated',
+          icon: 'edit',
+          color: 'orange',
+          sortable: false
+        },
+        { text: 'Size', value: 'project_size', sortable: false },
+        { text: '', value: 'archived', sortable: false }
+      ],
+      options: {
+        sortDesc: [true],
+        itemsPerPage: this.defaultItemsPerPage ?? 50,
+        page: 1
+      }
+    }
+  },
+  computed: {
+    ...mapState('projectModule', [
+      'versions',
+      'versionsLoading',
+      'versionsCount'
+    ]),
+    /**
+     * Table data from versions in global state transformed
+     */
+    items(): ProjectVersionsItem[] {
+      const versions: ProjectVersion[] = this.versions
+
+      return versions?.map<ProjectVersionsItem>((v) => ({
+        ...v,
+        disabled: this.disabledKeys.some((d) => d === v.name)
+      }))
+    }
+  },
+  watch: {
+    $route: 'fetchVersions'
+  },
+  methods: {
+    ...mapActions('projectModule', ['fetchProjectVersions']),
+    paginating(options) {
+      this.options = options
+      this.fetchVersions()
+    },
+    fetchPage(number) {
+      this.options.page = number
+      this.fetchVersions()
+    },
+    disabledRow(item: ProjectVersionsItem) {
```

Comment:
I would rename this to something more suitable:
* I would add Style suffix as it defines value for "style" and not value for "row".
* It is disabled only when `item.disabled` is `true `- hence it could be something like `disabledOrEnabledRowStyle`